resource.go

```diff
@@ -58,8 +62,8 @@ type Resource struct {
 }
 
 func New(config Config) (*Resource, error) {
-	if config.ClusterEndpointFunc == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterEndpointFunc must not be empty", config)
+	if config.BackOffFactory == nil {
+		config.BackOffFactory = func() backoff.Interface { return backoff.NewMaxRetries(3, 1*time.Second) }
```

Comment:
I am a bit confused how long we want to retry here. This seems a little short now but no strong feelings.

Reply:
We have also retryresource, check the issue linked in the desc.

Reply:
This should be fine.